### ISSUE TITLE
Angel One HLS asset failures

### ISSUE THREAD

**Issue by theodab** [google/shaka-player#1137]
On the nightly, the Angel One HLS assets are both behaving incorrectly.
The clear one seems to infer the start time as being ~8.5 seconds before the real start time, which results in there being a large gap at the start of the video.
In a possibly-related note, the Widevine-encrypted version outputs a BUFFER_READ_OUT_OF_BOUNDS error with the following stack trace:
```
Error: Shaka Error MEDIA.BUFFER_READ_OUT_OF_BOUNDS ()
    at new shaka.util.Error (https://nightly-dot-shaka-player-demo.appspot.com/lib/util/error.js:72:13)
    at shaka.util.DataViewReader.throwOutOfBounds_ (https://nightly-dot-shaka-player-demo.appspot.com/lib/util/data_view_reader.js:288:9)
    at shaka.util.DataViewReader.skip (https://nightly-dot-shaka-player-demo.appspot.com/lib/util/data_view_reader.js:223:10)
    at shaka.util.Mp4Parser.parseNext (https://nightly-dot-shaka-player-demo.appspot.com/lib/util/mp4_parser.js:236:12)
    at shaka.util.Mp4Parser.parse (https://nightly-dot-shaka-player-demo.appspot.com/lib/util/mp4_parser.js:161:10)
    at shaka.hls.HlsParser.getStartTimeFromMp4Segment_ (https://nightly-dot-shaka-player-demo.appspot.com/lib/hls/hls_parser.js:1348:10)
    at shaka.hls.HlsParser.<anonymous> (https://nightly-dot-shaka-player-demo.appspot.com/lib/hls/hls_parser.js:1304:19)
    at <anonymous>
```

This only happens on the nightly, so it's probably the result of a recent CL; we've had a number that have to do with time offset in HLS recently, so I'm not sure which.

**Comment by TheModMaker** [comment]
The clear asset hasn't played since we added Live support in 113d974.
The encrypted asset hasn't played since we added partial segment requests in a93455a9.

**Comment by joeyparrish** [comment]
The encrypted version is failing, too.  I'm looking into it.

**Comment by joeyparrish** [comment]
The encrypted version fails because we aren't fetching enough of the segments to find the tfdt box and extract the timestamp.

The clear version fails because the text stream starts at 0 and the video & audio streams start at 8+change, so the HLS parser sets presentationTimeOffset to 0.

**Comment by joeyparrish** [comment]
Fix cherry-picked for v2.2.7.
